py3status/modules/mpris.py

```diff
@@ -0,0 +1,378 @@
+            else:
+                players_stopped.append(player_name)
+
+        if players_paused:
+            return players_paused[0]
+        if players_stopped:
+            return players_stopped[0]
+
+        return 'None'
+
+    def _get_text(self, i3s_config):
+        """
+        Get the current metadatas
+        """
+        is_stream = False
+        player = UNKNOWN
+        state = UNKNOWN
+        artist = UNKNOWN
+        album = UNKNOWN
+        length = UNKNOWN_TIME
+        time = UNKNOWN_TIME
+        title = UNKNOWN
+
+        try:
+            player = self._player.Identity
+            (color, state) = self._get_state_format(i3s_config)
+
+            _time_ms = self._player.Position
+            time = _get_time_str(_time_ms)
+
+            metadata = self._player.Metadata
+            if len(metadata) > 0:
+                is_stream = 'file://' not in metadata.get('xesam:url')
+                title = metadata.get('xesam:title') or UNKNOWN
+                album = metadata.get('xesam:album') or UNKNOWN
+
+                if metadata.get('xesam:artist') is not None:
+                    artist = metadata.get('xesam:artist')[0]
+                else:
+                    # we assume here that we playing a video and these types of
+                    # media we handle just like streams
+                    is_stream = True
+
+                _length_ms = metadata.get('mpris:length')
+                if _length_ms is not None:
+                    length = _get_time_str(_length_ms)
+            else:
+                # use stream format if no metadata is available
+                is_stream = True
+        except Exception:
+            color = i3s_config['color_bad']
+
+        if is_stream:
+            # delete the file extension
+            title = re.sub(r'\....$', '', title)
+            format = self.format_stream
+        else:
+            format = self.format
+
+        return (format.format(player=player,
+                              state=state,
+                              album=album,
+                              artist=artist,
+                              length=length,
+                              time=time,
+                              title=title), color)
+
+    def _get_response_buttons(self):
+        response = []
+
+        if self._player.PlaybackStatus == 'Playing':
+            state = self.state_pause
+        else:
+            state = self.state_play
+
+        buttons_order = self.buttons_order.split(',')
+        for button in buttons_order:
+            if button == 'play':
+                response.append({
+                    'color': '#CCCCCC',
+                    'full_text': ' ' + state + ' ',
+                    'index': 'play',
+                })
+            elif button == 'stop':
+                response.append({
+                    'color': '#CCCCCC',
+                    'full_text': ' ' + self.state_stop + ' ',
```

Comment:
`state_play`, `state_pause` and `state_stop` are also used for the placeholder `{state}`. If you move the spaces from the source code to the format string than you can be get an ugly text e.g. for `format = "{state} {title}"` you'll get `" ◾ We Will Rock You"`.

But yes I also don't like this code. Just couldn't find a better solution here…